### PR TITLE
Add Google Cloud Functions adapter (ref #646)

### PR DESCRIPTION
**Pull request by seratch**
This pull request adds a new adapter for Google Cloud Functions users. You can use Flask adapter for it as long as you don't have the OAuth flow in your app. However, as we see at #646,  implementing the OAuth flow is far different and it would be confusing for developers. This pull request adds a new adapter that is quite similar to the AWS Lambda adapter. 

One thing developers should be aware of would be the lazy listeners are not yet available for Google Cloud Functions, unlike AWS Lambda. Thus, I've added Noop runner, which just raises an exception when a developer tries to use lazy listeners in Cloud Functions app. Also, this is an improvement from the Flask adapter way.

### Category (place an `x` in each of the `[ ]`)

* [ ] `slack_bolt.App` and/or its core components
* [ ] `slack_bolt.async_app.AsyncApp` and/or its core components
* [x] Adapters in `slack_bolt.adapter`
* [ ] Document pages under `/docs`
* [ ] Others

## Requirements (place an `x` in each `[ ]`)

Please read the [Contributing guidelines](https://github.com/slackapi/bolt-python/blob/main/.github/contributing.md) and [Code of Conduct](https://slackhq.github.io/code-of-conduct) before creating this issue or pull request. By submitting, you are agreeing to those rules.

* [x] I've read and understood the [Contributing Guidelines](https://github.com/slackapi/bolt-python/blob/main/.github/contributing.md) and have done my best effort to follow them.
* [x] I've read and agree to the [Code of Conduct](https://slackhq.github.io/code-of-conduct).
* [x] I've run `./scripts/install_all_and_run_tests.sh` after making the changes.

### CODE REVIEW

**Comment by seratch** [comment]
Ah, yeah I should add some tests. Will do after finishing other tasks.

File: slack_bolt/adapter/google_cloud_functions/handler.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    def __init__(self, app: App):  # type: ignore
+        self.app = app
+        # Note that lazy listener is not supported
+        self.app.listener_runner.lazy_listener_runner = NoopLazyListenerRunner()
+        if self.app.oauth_flow is not None:
+            self.app.oauth_flow.settings.redirect_uri_page_renderer.install_path = "?"
+
+    def handle(self, req: Request) -> Response:
+        if req.method == "GET":
+            if self.app.oauth_flow is not None:
```
Comment (stasfilin) [inline slack_bolt/adapter/google_cloud_functions/handler.py:51]: We can remove this extra **if**

Just change 
```python
        if req.method == "GET":
            if self.app.oauth_flow is not None:
```

to 

```python
        if req.method == "GET" and self.app.oauth_flow is not None
```

Comment (seratch) [reply slack_bolt/adapter/google_cloud_functions/handler.py:51]: @stasfilin Thanks for the suggestion!

**Comment by seratch** [comment]
Updated the code (thanks for the suggestion, @stasfilin!) and added unit tests ✅

File: examples/google_cloud_functions/oauth_main.py [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+    @flask_app.route("/function", methods=["GET", "POST"])
+    def handle_anything():
+        return handler.handle(request)
+
+    flask_app.run(port=3000)
+
+
+# Step1: Create a new Slack App: https://api.slack.com/apps
+# Bot Token Scopes: chat:write, commands, app_mentions:read
```
Comment (filmaj) [inline examples/google_cloud_functions/oauth_main.py:83]: FYI the .env.yaml.oauth-sample has a different list than this; it includes `channels.history`.

**Review by filmaj — APPROVED** [review]
Code LGTM but I did not create a test app using this as salesforce does not allow me to create Google Cloud projects 😑

File: examples/google_cloud_functions/oauth_main.py (hunk 2 above)
Comment (seratch) [reply examples/google_cloud_functions/oauth_main.py]: Thanks! I've updated the part.

**Pull request merged**